src/interface/params.cpp

```diff
@@ -58,25 +58,19 @@ void Params::ReadFromHDF5AllParamsOfType(const std::string &prefix,
                                          const HDF5::H5G &group) {
   for (auto &[key, pparam] : myParams_) {
     if (std::type_index(pparam->type()) == std::type_index(typeid(T)) &&
-        IsRestartable(key)) {
-      auto mutability = myMutable_.at(key);
+        GetMutability(key) == Mutability::Restart) {
       auto typed_ptr = std::any_cast<T>(pparam.get());
       auto &val = *typed_ptr;
       std::string fullpath = prefix + "/" + key;
-      if (mutability == Mutability::Restart) {
+      try {
```

Comment:
I think we should discuss this particular item a bit more.... I am concerned that loading from a restart with incomplete data without erroring out will cause a lot of hard to find bugs down the line. It seems like it would be really easy for someone to trigger this without thinking too hard about it.